src/main/java/org/mockito/internal/matchers/CapturingMatcher.java

```diff
@@ -33,12 +33,20 @@ public CapturingMatcher(final Class<? extends T> clazz) {
 
     @Override
     public boolean matches(Object argument) {
```

Comment:
Nit: I think we can change `Object` to `T` here as well? It doesn't change the implementation of the method, but since we are `ArgumentMatcher<T>` here I think that should be safe. However, this is okay for now and we can consider this as part of the follow-up as well.